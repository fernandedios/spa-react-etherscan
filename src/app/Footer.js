import React from 'react';
import './../index.css';

const Footer = () => {
    return (
        <footer className="footer">
            <div className="container">
                <div className="content has-text-centered">
                    <p>SPA Etherscan App &copy; Fernan de Dios. Released under MIT</p>
                </div>
            </div>
        </footer>
    );
}

export { Footer };