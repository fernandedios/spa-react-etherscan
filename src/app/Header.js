import React from 'react';
import './../index.css';

const Header = () => {
    return (
        <nav className="navbar is-info">
            <div className="navbar-brand">
                <a className="navbar-item" href="/">ReactJS with Redux</a>
            </div>
        </nav>
    );
}

export { Header };
