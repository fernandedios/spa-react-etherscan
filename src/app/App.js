import React, { Component } from 'react';
import { Header } from './Header';
import { Footer } from './Footer';
import Search from '../search/Search';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Search />
        <Footer />
      </div>
    );
  }
}

export default App;
