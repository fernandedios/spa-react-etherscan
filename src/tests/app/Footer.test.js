import React from 'react';
import { shallow } from 'enzyme';
import ReactTestRenderer from 'react-test-renderer';
import { Footer } from '../../app/Footer';

describe('<Footer />', () => {
  it('should be defined', () => {
    expect(Footer).toBeDefined();
  });

  it('should have the correct classNames', () => {
    const output = shallow(<Footer />);

    expect(output.find('footer').hasClass('footer')).toEqual(true);
  });

  it('should render correctly', () => {
    const output = ReactTestRenderer.create(<Footer />).toJSON();
    expect(output).toMatchSnapshot();
  });
});
