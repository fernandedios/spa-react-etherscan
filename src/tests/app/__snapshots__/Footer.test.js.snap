// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <div
      className="content has-text-centered"
    >
      <p>
        SPA Etherscan App Â© Fernan de Dios. Released under MIT
      </p>
    </div>
  </div>
</footer>
`;
